### ISSUE TITLE
CloudFront doesn't pick up changes in Lambda version

### ISSUE THREAD

**Issue by jsonmaur** [terraform-providers/terraform-provider-aws#1124]
### Terraform Version
v0.9.11

### Affected Resource(s)
- aws_lambda_function
- aws_cloudfront_distribution

### Terraform Configuration Files
```hcl
resource "aws_lambda_function" "main" {
  filename = "${data.archive_file.main.output_path}"
  source_code_hash = "${data.archive_file.main.output_base64sha256}"
  publish = true
  # ...
}

resource "aws_cloudfront_distribution" "main" {
  default_cache_behavior {
    lambda_function_association {
      event_type = "origin-request"
      lambda_arn = "${aws_lambda_function.main.qualified_arn}"
    }
    # ...
  }
  # ...
}
```

### Expected Behavior
Any updates to the function source code should update the lambda function (and increment the version), then apply the new version to the CloudFront distribution in a single apply.

### Actual Behavior
It only updates the function during the first apply. You have to run `terraform apply` again to update the CF distro with the new function version.

Also, if you change the contents of the function two times in a row (before applying twice so it goes to the CF distro), Terraform crashes on the second apply.

### Steps to Reproduce
Run `terraform apply` twice.

**Comment by jbellmore** [comment]
I can confirm I am seeing similar behavior. Only difference I see is that even if I run terraform apply twice the cloudfront trigger does not get set on the new version of the lambda function. The only time I have seen the lambda cloudfront trigger get set was on the initial creation of the lambda function.

I can confirm I am seeing the error when running apply with two changes to the lambda function in a row.

**Comment by jbellmore** [comment]
Hi again, just want to give an update, it turns out my cloudfront triggers **were** being added like @jsonmaur was describing, however due to what I assume is a bug in the AWS console they do not appear in the "triggers" section in the lambda configuration but they **do** appear in the cloudfront  behavior in the console.

So I am seeing the exact same behavior as @jsonmaur.

Thanks!

**Comment by arkadiyt** [comment]
+1 also running into this issue

**Comment by bflad** [comment]
Thanks to @mdlavin the fix for this has been merged into master and will be released in v1.10.0 of the AWS provider, likely later today or Monday. 🎉

**Comment by bflad** [comment]
This has been released in [version 1.10.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1100-february-24-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
